File: baekjoon/14658/14658.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.ArrayList;
import java.util.List;
import java.util.StringTokenizer;

public class Main {
    private static final BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
    private static final BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
    private static StringTokenizer stringTokenizer;
    private static int N, M, L, K;
    private static List<Coordinate> coordinates = new ArrayList<>();
    private static int result;


    public static void main(String[] args) throws IOException {
        stringTokenizer = new StringTokenizer(reader.readLine());
        N = Integer.parseInt(stringTokenizer.nextToken());
        M = Integer.parseInt(stringTokenizer.nextToken());
        L = Integer.parseInt(stringTokenizer.nextToken());
        K = Integer.parseInt(stringTokenizer.nextToken());

        for (int i = 0; i < K; i++) {
            stringTokenizer = new StringTokenizer(reader.readLine());
            int x = Integer.parseInt(stringTokenizer.nextToken());
            int y = Integer.parseInt(stringTokenizer.nextToken());
            coordinates.add(new Coordinate(x, y));
        }

        solve();

        writer.write(String.format("%d\n", result));
        writer.flush();

        writer.close();
    }

    private static void solve() {
        int max = -1;

        for (Coordinate coordinate1 : coordinates) {
            for (Coordinate coordinate2 : coordinates) {
                int bouncingCount = getBouncingCount(coordinate1.x, coordinate2.y);
                if (bouncingCount > max) {
                    max = bouncingCount;
                    result = K - max;
                }
            }
        }
    }

    private static int getBouncingCount(int x, int y) {
        int boundingCount = 0;

        for (Coordinate coordinate : coordinates) {
            if (coordinate.x >= x && coordinate.x <= x + L && coordinate.y >= y && coordinate.y <= y + L) {
                boundingCount++;
            }
        }

        return boundingCount;
    }

    private static class Coordinate {
        private int x;
        private int y;

        public Coordinate(int x, int y) {
            this.x = x;
            this.y = y;
        }
    }

}

File: baekjoon/14658/14658.py
import sys

N = None
M = None
L = None
K = None
coordinates = []

def get_bouncing_count(x, y):
    bouncing_count = 0

    for coordinate in coordinates:
        if x <= coordinate[0] <= x + L and y <= coordinate[1] <= y + L:
            bouncing_count += 1

    return bouncing_count

def solve():
    result = None
    max = -1

    for coordinate1 in coordinates:
        for coordinate2 in coordinates:
            bouncing_count = get_bouncing_count(coordinate1[0], coordinate2[1])
            if bouncing_count > max:
                max = bouncing_count
                result = K - max

    return result

if __name__ == '__main__':
    N, M, L, K = map(int, sys.stdin.readline().split())

    for _ in range(K):
        x, y = map(int, sys.stdin.readline().split())
        coordinates.append((x, y))

    print(solve())
